presto-main/src/main/resources/webapp/src/components/SQLClient.jsx

```diff
@@ -0,0 +1,277 @@
+                        highlight={code => highlight(code, languages.sql)}
+                        padding={10}
+                        style={{
+                            fontFamily: '"Fira code", "Fira Mono", monospace',
+                            fontSize: 14,
+                            borderWidth: '2px',
+                            borderColor: 'black',
+                            border: 'solid',
+                            margin: '4px',
+                        }}
+                        placeholder='Type your SQL query...'
+                    />
+                </div>
+            </div>
+        </>
+    );
+}
+
+function QueryResults({results}) {
+    const customStyles = {
+        headCells: {
+            style: {
+                padding: '2px', // override the cell padding for head cells
+                fontSize: '15px',
+            },
+        },
+        cells: {
+            style: {
+                padding: '2px', // override the cell padding for data cells
+                fontSize: '15px',
+            },
+        },
+    };
+
+    const getColumns = () => {
+        return results.columns.map( (row, index) => {
+            return {
+                name: row.name,
+                selector: row => row[index],
+            };
+        });
+    };
+
+    return (
+        <>
+            <div className="row">
+                <div className='col-xs-6'>
+                    <h3>Results</h3>
+                </div>
+                <div className="col-xs-6">
+                    <a style={{display: 'block', marginTop: '27px', marginBottom: '10px'}}
+                        href={"query.html?" + results.queryId} target="_blank"
+                        data-toggle="tooltip" data-trigger="hover" title="Query ID"
+                    >
+                        {results.queryId}
+                    </a>
+                </div>
+            </div>
+            <div className="row"><hr className="h3-hr"/></div>
+            {results.error && <div className="row"><div className="alert alert-danger" role="alert"><h4 className="text-center">{results.error.message}</h4></div></div>}
+            {results.data && <div className="row"><DataTable columns={getColumns()} data={results.data} theme='dark' customStyles={customStyles} striped='true' pagination /></div>}
+        </>
+    );
+}
+
+export default function SQLClientView() {
+    const [values, setValues] = React.useState({sql: '', running: false, results: undefined});
+
+    const executeSQL = (sqlInfo) => {
+        const client = createClient(sqlInfo.catalog, sqlInfo.schema);
+          setValues({...values, running: true, results: undefined});
+          client.query(sqlInfo.sql).then((prestoQuery) => {
+              if (prestoQuery.data && prestoQuery.data.length > 0) {
+                setValues({...values, running: false, results: prestoQuery});
+              } else {
+                // presto-js-client doesn't return error
```

Comment:
this seems like a good potential improvement on the presto-js-client
I created an issue for it: https://github.com/prestodb/presto-js-client/issues/11

Reply:
awesome! You spotted it! I was thinking it would be good to get the queryId first and another call either pull the result or wait for the query to finish and get the whole results at once. But it's up to you and once you have new change in `presto-js-client` I can adopt it here.

Reply:
sounds good, I also think we could add a couple of extra things to the js-client:
- get catalogs/schemas utility
- get query metadata

eventually the web ui should be able to use the js-client for getting all of the things instead of having to do `fetch` manually to the REST API

Reply:
I am also curious why I don't get any error. Once you have any change in presto-js-client, please tag me and I can update the code here. Thanks!

Reply:
> sounds good, I also think we could add a couple of extra things to the js-client:
> 
>     * get catalogs/schemas utility
> 
>     * get query metadata

I like this idea! +1